code/game/objects/structures/watercloset.dm

```diff
@@ -453,7 +453,12 @@
 	icon_state = "sink"
 	desc = "A sink used for washing one's hands and face."
 	anchored = 1
-	var/busy = 0 	//Something's being washed at the moment
+	var/busy = 0 //Something's being washed at the moment
+	var/obj/item/weapon/reagent_containers/glass/beaker/water/watersource = null //The liquid dispensed by the sink, normally water, lifted from the shower
```

Comment:
There _has_ to be a better way to handle this than an invisible infinte (?) beaker
You also need to set this to qdel & set this to null in `Destroy()`

Reply:
That would be the solution the shower took, and yeah

Reply:
Make a `reagents` datum on the sink and use it directly 
Rather than using `my_beaker.reagents`